hibernate-types-4/src/main/java/com/vladmihalcea/hibernate/type/range/guava/PostgreSQLGuavaRangeType.java

```diff
@@ -81,7 +81,13 @@ protected void set(PreparedStatement st, Range range, int index, SessionImplemen
     }
 
     private static String determineRangeType(Range<?> range) {
-        Object anyEndpoint = range.hasLowerBound() ? range.lowerEndpoint() : range.upperEndpoint();
+        Object anyEndpoint = range.hasLowerBound() ? range.lowerEndpoint() :
+                             range.hasUpperBound() ? range.upperEndpoint() : null;
+
+        if (anyEndpoint == null) {
+            throw new IllegalArgumentException("Cannot find bound type");
```

Comment:
There might not be a range type here, but this case should probably just return `Range.all()`. The problem with `Range.all()` is that - although we can read it - we cannot write it, because we have no bound to infer the type (and thus the PostgreSQL type for the `PGObject` from). For that it would be nice if the type could be specified as a parameter of the `PostgreSQLGuavaRangeType` instead. Then `Range.all()` can just be mapped fine and we don't have to throw this dubious error.